syntax = "proto3";
package sqlpkg_versions;





/*
CreateSQLPackageVersion请求
*/
message CreateSQLPackageVersionRequest {
message CreateSqlPkgVersion {
    /*
    基线版本ID
    */
    string baseVersionId = 1 ;
    /*
    版本名称
    */
    string versionName = 2 ;
    /*
    备注
    */
    string memo = 3 ;
    /*
    仓库版本ID
    */
    string repoVersionId = 4 ;
}
        
    
    /*
    SQL包ID
    */
    string pkgId = 1 ;
    /*
    创建SQL包版本body
    */
    CreateSqlPkgVersion createSqlPkgVersion = 2 ;
}

/*
CreateSQLPackageVersion返回
*/
message CreateSQLPackageVersionResponse {
message SqlPackage {
    /*
    SQL包ID
    */
    string id = 1 ;
    /*
    名称
    */
    string name = 2 ;
    /*
    描述
    */
    string memo = 3 ;
    /*
    文件仓库包ID
    */
    string repoPackageId = 4 ;
    /*
    创建者
    */
    string creator = 5 ;
    /*
    mtime
    */
    int64 mtime = 6 ;
}
        
    
    /*
    所属的SQL包
    */
    SqlPackage sqlPackage = 1 ;
    /*
    版本ID
    */
    string id = 2 ;
    /*
    名称
    */
    string name = 3 ;
    /*
    仓库版本ID
    */
    string repoVersionId = 4 ;
    /*
    版本名称
    */
    string versionName = 5 ;
    /*
    备注
    */
    string memo = 6 ;
    /*
    创建者
    */
    string creator = 7 ;
    /*
    创建时间
    */
    int64 ctime = 8 ;
    /*
    修改时间
    */
    int64 mtime = 9 ;
    /*
    版本类型, [开发,测试,预发布,生产]
    */
    string versionType = 10 ;
}

/*
CreateSQLPackageVersionApi返回
*/
message CreateSQLPackageVersionResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    sqlpkg_versions.CreateSQLPackageVersionResponse data = 4 ;
}

